### DISCUSSION TITLE
empty parsed object

### DISCUSSION THREAD

**Discussion by kraabrsg** [NaturalIntelligence/fast-xml-parser#478] [Q&A]
Hi, sorry if i have overlooked this questions.

In our vscode extension we want to parse xmls, here is the simplified version.

![grafik](https://user-images.githubusercontent.com/86772851/180444114-e08e730b-e876-4029-ad5f-30e79fc550cd.png)

```
    const options = {
        ignoreDeclaration: true,
        trimValues: false,
        unpairedTags: ["rd_3"] 
    };
    const parser = new XMLParser(options);
    
    let jObj = parser.parse(xmlData);
```

We would like to get the key and value of "sql_3" and of the "included" "rd_3" ("NAME", "COLNO" etc...), but as can be seen on the screenshot "rd_3" is an empty string.

There are so many amazing options, it would be really nice if we could use this.

Thank you!

**Comment by amitguptagwl** [comment]
Can you please share some sample XML?

**Reply by kraabrsg** [reply]
Hi, 

well i tried it with an ordinary variable, like this (as can be seen on the screenshot too).

```
    let xmlData = ' <sql_3  id = "XXXXX.YYYYYY"> ' + 
      ' <rd_3  NAME="ZZZZ" COLNO="2" COLTYPE="CHAR    "  FOREIGNKEY="S" />' +
       '</sql_3>' ;
```

**Reply by amitguptagwl** [reply]
probably, I'm not understanding your goal. I can see `rd_3` is a normal tag but you have defined it as `unpairedTags`. So can it be both? what is the expected data for above XML?

**Reply by kraabrsg** [reply]
Well, that is an xml we read, we have to deal with it.

Unfortunatly github won't let me upload screenshots ATM.

it should get a asturcture:

```
tempobj: {..., sql_3: Array(2) , ...}
sql_3:  (2) [ {...}, {...} ]
   0: {$: {..}, rd_3: Array(12) }
        $: { xxx: 'yyyy'}
             xxx: 'yyy'
             aaaa: 'bbb'
        rd_3: (12) [ {..} , {..}, ...]
            0: {$: {...}}
                $: {NAME: 'zzz', ...}
```

so accessing it( key value pairs in $, e.g. NAME and 'zzz'  is possible ) with e.g. 
```
tempObj['sql_3'][0][0]['$'] 
```

We tried it with xml2js, where it seems to work, like described above...
